distribution/lib/Standard/Visualization/0.0.0-dev/src/Helpers.enso

```diff
@@ -136,9 +138,18 @@ Vector.default_visualization self = Id.table
    Transform the vector into text for displaying as part of its default
    visualization.
 Vector.to_default_visualization_data : Text
-Vector.to_default_visualization_data self =
-    json = self.take 100 . to_json
-    json.to_text
+Vector.to_default_visualization_data self = render_vector self
+
+render_vector object depth=0 =
+    case object of
+        _ : Vector -> if depth == 5 then "[...]" else
+            result = object.take 100 . map (elem-> if elem.is_error then elem.to_json else render_vector elem (depth+1))
+            "[" + result.join ", " + "]"
+        _ : Array ->
+            proxy = Array_Proxy.new object.length object.at
+            @Tail_Call render_vector (Vector.from_polyglot_array proxy) depth
+        _ : JS_Object -> object.to_default_visualization_data depth
```

Comment:
Personally, I don't like that `JS_Object` has `to_default_visualization_data` taking additional arguments. While the API is more or less the same as they have defaults, it feels like a potential source of issues. I'd just delegate to `Json.render` private method here instead.